### DISCUSSION TITLE
How do I dynamically change the image from the file system

### DISCUSSION THREAD

**Discussion by mike-ensight** [unoplatform/uno#15116] [Q&A] [answered]
Hello, I am developing an app UI that only supports Linux and Windows, so I have a GTK project. I have read the articles on how to load and show images, like this one: [Working with Images](https://platform.uno/docs/articles/features/working-with-assets.html). The problem is, most of the articles and examples like this use the Assets folder, which means I need to load the images up front, at compile time. 

In my case, I need to do this dynamically, from the file system. Think of it like a photo viewer that is common to some applications. You click on a button to select the image file and then that file shows on the UI. 

If I have the path of the image on the file system, how can I show it on the UI? I am using the <Image> tag at the moment but I am fine changing that, just remember, I need just Linux and WIndows. 

Thanks!

**Comment by jeromelaban** [comment, marked as answer]
You can use [`BitmapImage.SetSourceAsync()`](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.media.imaging.bitmapsource.setsourceasync?view=winrt-22621) to set the image if it comes from a path outside your app's files.

You can also use the `ms-appdata:///` scheme to load images located in the local app's data folder `ApplicationData.Current.LocalFolder`.
